library/Core/test/Conv.hs

```diff
@@ -48,7 +56,11 @@ hrToirConversion =
               IR.Bound 0
                 `IR.App` IR.Lam (IR.Elim $ IR.Bound 0)
                 `IR.App` IR.Lam (IR.Elim $ IR.Free (IR.Global "x"))
-        )
+        ),
+      shouldConvertHRWith
+        [HR.PVar "a", HR.PVar "hi"]
+        (HR.Elim (HR.Var "a"))
+        (IR.Elim (IR.Free (IR.Pattern 0)))
```

Comment:
It'd be nice to test your change, where you have also managed to update the counter!